lib/block-patterns.php

```diff
@@ -228,12 +209,16 @@ function() {
 			unregister_block_pattern( 'core/' . $core_block_pattern );
 		}
 
-		foreach ( $new_core_block_patterns as $core_block_pattern ) {
-			register_block_pattern(
-				'core/' . $core_block_pattern,
-				require __DIR__ . '/block-patterns/' . $core_block_pattern . '.php'
-			);
+		$request = new WP_REST_Request( 'GET', '/__experimental/pattern-directory/patterns' );
+		$request->set_param( 'keyword', 11 ); // 11 is the ID for "core".
```

Comment:
Just noting that without the comment this would be impossible to understand since the keyword ID lives outside of this project. Additionally, if any changes happen in the directory, this would return incorrect patterns. I think for now it's ok, but if core patterns maintain priority in the future, we may want to consider a different approach to loading core patterns that doesn't rely on a hard coded id.

Reply:
That's why I left the comment 🙂 

This really is a "for now" kind of feature — to move the management of these core patterns into the block pattern directory. The longer-term plan is for a more dynamic search UI that fetches from the API on the client side (like the block directory), any prioritization would happen on the API side, so Gutenberg wouldn't need to be aware of this category at all.

Reply:
Can we make that `11` be a variable with a good name like `$core_keyword_id = 11` and also have that comment?

Reply:
It's a taxonomy on wp.org/patterns. Is that a problem? I was trying to make the least changes to the wp.org api, but this could probably be updated if so. cc @iandunn

Reply:
It's been awhile, but if I'm reading everything correctly, I think we could rename the `keyword` param to `keyword_id` (or maybe plural), and it'd only effect `WP_REST_Pattern_Directory_Controller::get_items()`.

On the API side it's `pattern-categories`, and `get_items()` maps between the two.

Using a slug would be nice, but IIRC the REST API only accepts IDs, unless we add custom code. I haven't checked that, though.

Either way, IMO it's best to be practical in this case. As long as variable names and other easily updated things are clear, I think that's good enough. No need to get bogged down over something small.